toffy/watcher_callbacks.py

```diff
@@ -148,9 +171,33 @@ def generate_qc(self, qc_out_dir: str, panel: pd.DataFrame = None, **kwargs):
         for metric_name, data in metric_data.items():
             data.to_csv(os.path.join(qc_out_dir, metric_name), index=False)
 
+    def generate_mph(self, mph_output_dir, panel: pd.DataFrame = None):
+        """Generates mph metrics from given panel, and saves output to provided directory
+
+        Args:
+            mph_output_dir (str): where to output mph csvs to
+            panel (pd.DataFrame):
+                Target mass integration ranges
+
+        """
+
+        if self.__fov_data is None:
+            if panel is None:
+                raise ValueError('Must provide panel if fov data is not already generated...')
+            self._generate_fov_data(panel)
+
+        compute_mph_metrics(
+            bin_file_path=self.run_folder,
+            csv_dir=mph_output_dir,
+            fov=self.point_name,
+            target=self.__panel['Target'],
```

Comment:
Does this run? `panel['Target']` is going to be a list of all of the channels in the panel, right?